src/wire/icmp.rs

```diff
@@ -0,0 +1,58 @@
+use super::icmpv4;
+use super::icmpv6;
+
+macro_rules! v4v6 {
+    ( $class:ident ) => {
+        /// icmpv4::$class or icmpv6::$class
+        #[derive(Clone, PartialEq, Debug)]
+        pub enum $class {
+            #[cfg(feature = "proto-ipv4")]
+            Ipv4(icmpv4::$class),
+            #[cfg(feature = "proto-ipv6")]
+            Ipv6(icmpv6::$class),
+        }
+
+        impl From<icmpv4::$class> for $class {
+            fn from(s: icmpv4::$class) -> Self {
+                $class::Ipv4(s)
+            }
+        }
+        impl From<icmpv6::$class> for $class {
+            fn from(s: icmpv6::$class) -> Self {
+                $class::Ipv6(s)
+            }
+        }
+    };
+    ( $class:ident < $typearg:tt > ) => {
```

Comment:
This is the only matching arm that is used right? Or am I missing something?

Reply:
Yes, now it is. I think I'll remove the macro altogether, as there is only one class left.